pkg/kubelet/kubelet.go

```diff
@@ -1438,6 +1449,12 @@ func (kl *Kubelet) GetPodStatus(podFullName string, uid types.UID) (api.PodStatu
 
 	info, err := dockertools.GetDockerPodInfo(kl.dockerClient, manifest, podFullName, uid)
 
+	for _, c := range manifest.Containers {
+		containerStatus := info[c.Name]
+		containerStatus.Ready = kl.readiness.IsReady(strings.TrimPrefix(containerStatus.ContainerID, "docker://"))
```

Comment:
A container without a readiness probe should be declared ready if and only if it is in the running phase.